gapic/schema/api.py

```diff
@@ -358,7 +402,7 @@ def _get_fields(self, field_pbs: List[descriptor_pb2.FieldDescriptorProto],
                 enum=self.all_enums.get(field_pb.type_name.lstrip('.')),
                 message=self.all_messages.get(field_pb.type_name.lstrip('.')),
                 meta=metadata.Metadata(
-                    address=address,
+                    address=address.child(field_pb.name, path + (i,)),
```

Comment:
The purpose of this change was to have a correct understanding of the relative position of proto messages in a file.

In order to do this, I refactored `Address` -- in the _status quo ante_, the `address` object for a given thing knew what contained it, but not its position. You can think of it like having the street address but not the house number. This adds the position of them item as a first-class part of the address (adding the house number) so that the address paths are fully comparable and I can know whether one message comes before another.

The purpose of this is to not refer to messages that are not instantiated yet in proto files.

Reply:
> I see. I guess my understanding was that in cases like this the more appropriate thing to do would be to alias the import:

> ```python
> from gapic.schema import naming as _naming
> ```

You _can_ do that. It is not a requirement.